tests/integration/integration-test.sh

```diff
@@ -1,238 +1,305 @@
-#! /bin/bash
-set -euo pipefail
+set -eo pipefail
 
 # This script is expected to be called from the root folder of Connaisseur
-declare -i NUMBER_OF_VALID_DEPLOYMENTS=0
-declare -i NUMBER_OF_INVALID_DEPLOYMENTS=0
-
-echo 'Preparing Connaisseur config...'
-envsubst < tests/integration/update.yaml > update
-yq eval-all --inplace 'select(fileIndex == 0) * select(fileIndex == 1)' helm/values.yaml update
-rm update
-echo 'Config set'
-
-echo 'Installing Connaisseur...'
-# 'make' is chosen deliberately below to test the Makefile, while other tests use 'helm' directly
-make install || { echo 'Failed to install Connaisseur'; exit 1; }
-echo 'Successfully installed Connaisseur'
-
-echo 'Testing unsigned image...'
-kubectl run pod --image=securesystemsengineering/testimage:unsigned >output.log 2>&1 || true
-NUMBER_OF_INVALID_DEPLOYMENTS+=1
-
-if [[ ! "$(cat output.log)" =~ 'Unable to find signed digest for image docker.io/securesystemsengineering/testimage:unsigned.' ]]; then
-  echo 'Failed to deny unsigned image or failed with unexpected error. Output:'
-  cat output.log
+declare -A DEPLOYMENT_VALIDITY=(["VALID"]="0" ["INVALID"]="0")
+RED="\033[0;31m"
+GREEN="\033[0;32m"
+NC="\033[0m"
+SUCCESS="${GREEN}SUCCESS${NC}"
+FAILED="${RED}FAILED${NC}"
+EXIT="0"
+WOLIST=("CronJob" "DaemonSet" "Deployment" "Job" "Pod" "ReplicaSet" "ReplicationController" "StatefulSet")
+
+### SINGLE TEST CASE ####################################
+single_test() { # ID TXT TYP IMG NS OUT VAL
+  echo -n "[$1] $2"
+  if [[ "$3" == "deploy" ]]; then
+    kubectl run pod-$1 --image="$4" --namespace="$5" >output.log 2>&1 || true
+  elif [[ "$3" == "workload" ]]; then
+    envsubst <tests/integration/workload-objects/$4.yaml | kubectl apply -f - >output.log 2>&1 || true
+  else
+    kubectl apply -f $4 >output.log 2>&1 || true
+  fi
+  if [[ ! "$(cat output.log)" =~ "$6" ]]; then
+    echo -e ${FAILED}
+    echo "::group::Output"
+    cat output.log
+    kubectl logs -n connaisseur -lapp.kubernetes.io/instance=connaisseur
+    echo "::endgroup::"
+    EXIT="1"
+  else
+    echo -e "${SUCCESS}"
+  fi
+
+  if [[ $7 != "null" ]]; then
+    DEPLOYMENT_VALIDITY[$7]=$((${DEPLOYMENT_VALIDITY[$7]} + 1))
+  fi
+}
+
+### MULTI TEST CASE FROM FILE ####################################
+multi_test() {
+  test_cases=$(yq e -o=json ".test_cases.$1" tests/integration/cases.yaml)
```

Comment:
If the initial set is in yaml, why convert to JSON and use jq instead of yq on initial set?

Reply:
processing all in `yq` was very slow. `jq` was faster so i decided to output as json and work with `jq` instead.

Reply:
That's pretty unintuitive. Please add a comment to explain